libpod/container_internal_linux.go

```diff
@@ -668,11 +668,30 @@ func (c *Container) setupSystemd(mounts []spec.Mount, g generate.Generator) erro
 		}
 		g.AddMount(systemdMnt)
 	} else {
+		var statfs unix.Statfs_t
+		if err := unix.Statfs("/sys/fs/cgroup/systemd", &statfs); err != nil {
+			return errors.Wrapf(err, "/sys/fs/cgroup/systemd is not mounted on the host")
+		}
+
+		mountOptions := []string{"bind"}
+		if statfs.Flags&unix.MS_NODEV == unix.MS_NODEV {
+			mountOptions = append(mountOptions, "nodev")
+		}
+		if statfs.Flags&unix.MS_NOEXEC == unix.MS_NOEXEC {
+			mountOptions = append(mountOptions, "noexec")
+		}
+		if statfs.Flags&unix.MS_NOSUID == unix.MS_NOSUID {
+			mountOptions = append(mountOptions, "nosuid")
+		}
+		if statfs.Flags&unix.MS_RDONLY == unix.MS_RDONLY {
+			mountOptions = append(mountOptions, "ro")
+		}
+
 		systemdMnt := spec.Mount{
 			Destination: "/sys/fs/cgroup/systemd",
 			Type:        "bind",
 			Source:      "/sys/fs/cgroup/systemd",
-			Options:     []string{"bind", "nodev", "noexec", "nosuid", "rprivate"},
+			Options:     mountOptions,
```

Comment:
I think we should probably force `rprivate` even if not used by the host - @giuseppe @rhatdan Agree?

Reply:
Yes.  We should always add rprivate